### PR TITLE
Fixed files to no longer break rule of demeter

### PR DESCRIPTION
**Pull request by avjohnston**
Let's hope this works!

### CODE REVIEW

File: lib/item_repository.rb [hunk 1]
```diff
@@ -85,4 +85,8 @@ def delete(id)
     @items.delete(find_by_id(id))
   end
 
+  def item_count
+    all
+  end
```
Comment (DougWelchons) [inline lib/item_repository.rb]: we don't need this method at all. we can remove this and simply call `merchants.all` from our sales_engine

File: lib/merchant_repository.rb [hunk 2]
```diff
@@ -58,4 +58,5 @@ def update(id, attribute)
   def delete(id)
     @merchants.delete(find_by_id(id))
   end
+
```
[the remark this answers is not in the document]
Comment (DougWelchons) [reply lib/merchant_repository.rb:61]: if we are making other changes, can we remove this line? make the hound happy?

[the remark this answers is not in the document]
Comment (brisag) [reply lib/merchant_repository.rb:61]: Looks great!

**Comment by brisag** [comment]
Looks Great!

**Review by DougWelchons — COMMENTED** [review]
Over all looks good. I think we can merge. at some point we are going to have to integrate what I was working on as well to move the making of items and merchants into their repos.

**Pull request merged**